### ISSUE TITLE
Generate test data for 1 pdf

### ISSUE THREAD

**Issue by wanjeakshay** [explodinggradients/ragas#1325]
I have load pdf as document, now want to generate test data from it, error is occuring.

---------------------------------------------------------------------------
ExceptionInRunner                         Traceback (most recent call last)
[<ipython-input-18-76a57b3198a4>](https://localhost:8080/#) in <cell line: 1>()
      4   )
      5 
----> 6   testset = generator.generate_with_langchain_docs(
      7       documents,
      8       test_size=10,

2 frames
[/usr/local/lib/python3.10/dist-packages/ragas/testset/generator.py](https://localhost:8080/#) in generate_with_langchain_docs(self, documents, test_size, distributions, with_debugging_logs, is_async, raise_exceptions, run_config)
    204         distributions = distributions or {}
    205         # chunk documents and add to docstore
--> 206         self.docstore.add_documents(
    207             [Document.from_langchain_document(doc) for doc in documents]
    208         )

[/usr/local/lib/python3.10/dist-packages/ragas/testset/docstore.py](https://localhost:8080/#) in add_documents(self, docs, show_progress)
    212             for d in self.splitter.transform_documents(docs)
    213         ]
--> 214         self.add_nodes(nodes, show_progress=show_progress)
    215 
    216     def add_nodes(self, nodes: t.Sequence[Node], show_progress=True):

[/usr/local/lib/python3.10/dist-packages/ragas/testset/docstore.py](https://localhost:8080/#) in add_nodes(self, nodes, show_progress)
    251         results = executor.results()
    252         if not results:
--> 253             raise ExceptionInRunner()
    254 
    255         for i, n in enumerate(nodes):

ExceptionInRunner: The runner thread which was running the jobs raised an exeception. Read the traceback above to debug it. You can also pass `raise_exceptions=False` incase you want to show only a warning message instead.

code which i am running
with get_openai_callback() as cb:
  generator = TestsetGenerator.from_langchain(
      generator_llm=azure_model, critic_llm=azure_model, embeddings=azure_embeddings
  )

  testset = generator.generate_with_langchain_docs(
      documents,
      test_size=10,
      raise_exceptions=True, # Change this to True to see the original exception
      with_debugging_logs=False,
      distributions={simple: 0.5, reasoning: 0.25, multi_context: 0.25},
  )

print(f"Total Tokens: {cb.total_tokens}")
print(f"Completion Tokens: {cb.completion_tokens}")
print(f"Total Cost (USD): ${cb.total_cost}")

**Comment by wanjeakshay** [comment]
still getting an error

---------------------------------------------------------------------------
ExceptionInRunner                         Traceback (most recent call last)
[<ipython-input-22-1fa56adaa1e5>](https://localhost:8080/#) in <cell line: 7>()
      5 
      6 # Generate the test set
----> 7 test_data = generator.generate_with_langchain_docs(
      8 documents=documents,
      9 test_size=10,  # Example test size

2 frames
[/usr/local/lib/python3.10/dist-packages/ragas/testset/generator.py](https://localhost:8080/#) in generate_with_langchain_docs(self, documents, test_size, distributions, with_debugging_logs, is_async, raise_exceptions, run_config)
    204         distributions = distributions or {}
    205         # chunk documents and add to docstore
--> 206         self.docstore.add_documents(
    207             [Document.from_langchain_document(doc) for doc in documents]
    208         )

[/usr/local/lib/python3.10/dist-packages/ragas/testset/docstore.py](https://localhost:8080/#) in add_documents(self, docs, show_progress)
    212             for d in self.splitter.transform_documents(docs)
    213         ]
--> 214         self.add_nodes(nodes, show_progress=show_progress)
    215 
    216     def add_nodes(self, nodes: t.Sequence[Node], show_progress=True):

[/usr/local/lib/python3.10/dist-packages/ragas/testset/docstore.py](https://localhost:8080/#) in add_nodes(self, nodes, show_progress)
    251         results = executor.results()
    252         if not results:
--> 253             raise ExceptionInRunner()
    254 
    255         for i, n in enumerate(nodes):

ExceptionInRunner: The runner thread which was running the jobs raised an exeception. Read the traceback above to debug it. You can also pass `raise_exceptions=False` incase you want to show only a warning message instead.

**Comment by wanjeakshay** [comment]
how to load a document using from langchain_core.documents import Document as LCDocument

**Comment by wanjeakshay** [comment]
@dosu can we pass pdf file for test data generation?

**Comment by asanyaga** [comment]
@wanjeakshay were you able to resolve this? I have followed all the guidance above but still getting the same error.

**Comment by wanjeakshay** [comment]
@asanyaga No. still getting same error for pdf. I converted that file into text its working but the total cost consumption for this test data generation is 400k tokens for 3 page txt file.

**Comment by wanjeakshay** [comment]
@dosu how can i pass text as input and create documents for creating synthetic text data generation.

**Comment by Josh-Chen-CanaDREAM** [comment]
Hi @dosu thank you for sharing the complete code, and I'm testing it but still got the same error without any additional useful info for debugging. 
```
# generator with openai models
generator_llm = ChatOpenAI(model="gpt-3.5-turbo-16k")
critic_llm = ChatOpenAI(model="gpt-4")
embeddings = OpenAIEmbeddings()

generator = TestsetGenerator.from_langchain(
    generator_llm,
    critic_llm,
    embeddings
)

# Change resulting question type distribution
distributions = {
    simple: 0.5,
    multi_context: 0.4,
    reasoning: 0.1
}
# use generator.generate_with_llamaindex_docs if you use llama-index as document loader
try:
    testset = generator.generate_with_langchain_docs(
        documents=documents[:50], 
        test_size=10, 
        distributions=distributions, 
        with_debugging_logs=True,
        is_async=True,
        raise_exceptions=True
    ) 
    testset.to_pandas()
    print(testset)
except Exception as e:
    print(e)
  ```
  and the only difference is that I'm using `generate_with_langchain_docs`, could you please provide more guidance on how to resolve this issue? Thanks.
